airflow/providers/openlineage/extractors/bash.py

```diff
@@ -0,0 +1,72 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from airflow.providers.openlineage.extractors.base import BaseExtractor, OperatorLineage
+from airflow.providers.openlineage.plugins.facets import (
+    UnknownOperatorAttributeRunFacet,
+    UnknownOperatorInstance,
+)
+from airflow.providers.openlineage.utils.utils import is_source_enabled
+from openlineage.client.facet import SourceCodeJobFacet
+
+"""
+:meta private:
+"""
+
+
+class BashExtractor(BaseExtractor):
+    """
+    This extractor provides visibility on what bash task does by extracting
+    executed bash command and putting it into SourceCodeJobFacet. It does not extract
+    datasets.
+
+    :meta private:
+    """
+
+    @classmethod
+    def get_operator_classnames(cls) -> list[str]:
+        return ["BashOperator"]
```

Comment:
This is maybe a bigger question, but is there a reason this couldn't be just a property?

```suggestion
    operator_classnames: list[str] = ["BashOperator"]
```

Reply:
Could be, but we'd have to keep `get_operator_classnames` on base extractor for backwards compatibility.